hpx/compute/cuda/value_proxy.hpp

```diff
@@ -108,23 +103,10 @@ namespace hpx { namespace compute { namespace cuda
           , target_(other.target())
         {}
 
-        // Note: The code duplication below allows Cuda Clang to not see
-        //       the host side implicit cast during the device code
-        //       compilation. Without this, wrong template parameter
-        //       deduction can occur if a value_proxy is given in the arg
-        //       list of invoke()
-
-#if defined(__CUDA_ARCH__)
-        HPX_DEVICE operator T() const
+        HPX_HOST_DEVICE operator T() const
         {
             return access_target::read(target_, p_);
         }
```

Comment:
Was thinking on that ... The reason was the way `traits::access_target<>` has been designed in https://github.com/STEllAR-GROUP/hpx/blob/master/hpx/compute/cuda/traits/access_target.hpp